po/af.po

```diff
@@ -20,8 +20,8 @@ msgstr ""
 "X-Generator: Weblate 4.13-dev\n"
 
 #: data/com.github.wwmm.easyeffects.desktop.in:3
-msgid "EasyEffects"
-msgstr "EasyEffects"
+msgid "@APP_NAME@"
```

Comment:
I missed this in my PR… @wwmm any idea how to fix this?

We need to make the app name dyanamic so we can set it to `EasyEffects` or `EasyEffects (Devel)`. Should translators even be translating the app name in the first place though?

Reply:
>I missed this in my PR… @wwmm any idea how to fix this?

That is strange. I would expect Meson to first configure the file and only then to run the code related to the translation template. But it seems to be happening the contrary...

>Should translators even be translating the app name in the first place though?

It is hard to say. Some applications have their names translated. Others don't.

Reply:
> That is strange. I would expect Meson to first configure the file and only then to run the code related to the translation template. But it seems to be happening the contrary...

Indeed, it seems a bit strange meson behaves like this. as workaround maybe we can just `ninja` the metainfo file (and any others) before trying to update the templates in the update script?

> It is hard to say. Some applications have their names translated. Others don't.

It seems we can set`translatable="no"` in the name opening xml bracket.

I recall looking at the generated desktop file most languages except one don’t translate the name anyhow.

Reply:
>As a workaround maybe we can just ninja whatever build targets exist for the data directory before trying to update the templates in the update script?

But won't this happen anyway? I imagine it is like ninja install. These parts of the Meson scripts are probably being executed. Let me see...

>It seems we can settranslatable="no" in the name opening xml bracket.

The problem is that if we start to show the users `Easy Effects` instead of `EasyEffects` as it was suggested by the gnome circle people it makes sense to allow translations.  This is done to some gnome apps.

Reply:
>But won't this happen anyway? 

No it doesn't. I did not see this coming... We have to run `ninja` for the desktop file to be processed.

Reply:
>No it doesn't. I did not see this coming... We have to run ninja for the desktop file to be processed.

Nevermind. I did not test properly. After deleting the build folder and starting from a clean one the file was recreated even without calling ninja first.

Reply:
> Nevermind. I did not test properly. After deleting the build folder and starting from a clean one the file was recreated even without calling ninja first.

So should it be working correctly? All the action does is make the build folder then runs the script. 

https://github.com/wwmm/easyeffects/blob/208711164ddb35c3d78c6ec3fbd909fd55c92d2c/.github/workflows/translation-templates.yaml#L31

Reply:
>So should it be working correctly? 

I think it should. The correct file is created in the build folder. The i18n module just seems to be somehow accessing it before the tag replacement. What is weird.

Reply:
Since I have to guess perhaps it is because the intermittent outputted file that has the variable inputted shares the same name as the original? https://github.com/wwmm/easyeffects/blob/208711164ddb35c3d78c6ec3fbd909fd55c92d2c/data/meson.build#L62

I did that because `.in.in` look strange. But maybe having the file keep the name is confusing the translations.

Reply:
>Since I have to guess perhaps it is because the intermittent outputted desktop file that has the variable inputted shares the same name as the original? 

This crossed my mind. I will test if this changes anything.

Reply:
It seems more complicated. I tried to rename the desktop file to `com.github.wwmm.easyeffects.desktop.in.in`. But then I get the error
```
xgettext: error while opening "data/com.github.wwmm.easyeffects.desktop.in" for reading: No such file or directory
```
. But the file is there in the build folder. `xgettext` seems to be trying to get the one define in `POTFILES.in` before the code that process it is called.

Reply:
Oh... I think I see what is going on. The paths in `POTFILES.in` are to the files in our source folder. But the processed file is in the build folder. The code that is actually doing the translation does not see the processed file. It uses the raw file. I do not know if we can change that...

Reply:
If it is not possible to get meson/the translation code to use the built file as a lousy hack we could just edit the source file in the script manually, or maybe copy the file around…

Reply:
>If it is not possible to get meson/the translation code to use the built file as a lousy hack we could just edit the source file in the script manually, or maybe copy the built file around…

Although we could add the generated file to the `.gitignore` the idea of having the build system throwing things in the source folder does not seem good. But at the same time as the POTFILE is given as argument to `xgettext` I have my doubts we can make variable substitution to work with it.

Reply:
For now I will hardcode the name in the desktop file to avoid people changing the translation in weblate. Once we have a decent solution we try to name substitution in the desktop file again.

Reply:
Or maybe we just decide to forbid the translation of the application name. @Digitalone1 how to you feel about this? I see some gnome applications allow translations to their name. But many others do not do this. Firefox is one example.

Reply:
In the Firefox desktop file they translate only the `generic name`.

Reply:
To me the name has to be universal. The description and the comments have to be translated, but not the name.

Reply:
In this case I think we should do the same as Firefox and apply the solution @vchernin suggested:
>It seems we can settranslatable="no" in the name opening xml bracket.
How exactly is this done?

Reply:
That comment is wrong, I was reading the AppStream documention which is xml. I think we can still do a comment in the desktop file like:
`# Translators: This is a application name, don't translate!`

Reply:
Actually that xml comment is still useful. We should set the name https://github.com/wwmm/easyeffects/blob/58c793770a9618ede73414b975941b68d9416479/po/news/easyeffects-news.pot#L21 in the metainfo file to be non translatable. And probably wherever else the po files are finding `EasyEffects`.

Reply:
Ok. I am not sure I will have time to write code today. So feel free to open a PR.

Reply:
@vchernin So I have to copy and paste `@APP_NAME@` for Italian translation, right? Is it safe to do so?

If yes, I'll just copy for all languages and set the ignore flag for weblate warning, so it should not be touched afterwards.

Reply:
Yes, there is no reasons for translators to modify a variable like `@APP_NAME@`, so please set the ignore flag so it is not touched.

Reply:
One possibility would be to have two mostly identical desktop files, one for the regular app name, and one for the devel app name. Simply select which one you want to use as an input to `i18n.merge_file()` depending on which mode you are building in, and...

... list both files in POTFILES.

Reply:
I prefer to have less duplicate files, and given we decided it's best to not translate the app name in the first place, and since in weblate the variable can be set to not be touched I think this is fine.